subprojects/provider/src/main/kotlin/org/gradle/kotlin/dsl/provider/KotlinScriptEvaluator.kt

```diff
@@ -277,6 +283,27 @@ class StandardKotlinScriptEvaluator(
 }
 
 
+private
+fun Settings.addKotlinEapRepository() {
+
+    gradle.settingsEvaluated {
+        if (pluginManagement.repositories.isEmpty()) {
```

Comment:
:x: the following would cover more cases:

```kotlin
pluginManagement.repositories.run {
    val wasEmpty = isEmpty()
    kotlinEap()
    if (wasEmpty) {
        gradlePluginPortal()
    }
}
```

with the remaining caveat that a build that already has kotlin-eap will have it twice, in line with what is done for project and their buildscript below.